### ISSUE TITLE
Files with long name continue to next line

### ISSUE THREAD

**Issue by mfederczuk** [asuender/cliex#15]
Files with a very long name wrap and continue in the next line.  
After selecting a different file, the wrapped part still stays since the window doesn't get cleared.

![image](https://user-images.githubusercontent.com/14075939/82742384-b7d60f80-9d5d-11ea-981a-04dd3a833f44.png)
![image](https://user-images.githubusercontent.com/14075939/82742410-05527c80-9d5e-11ea-8d43-e0b3b8cbda8f.png)

**Comment by asuender** [comment]
This bug appears because `file_info_win()` only gets cleared at specific lines (where the information are shown) and not the whole window. Easy fix.

**Comment by mfederczuk** [comment]
Yeah, but we should actually look for a proper way to fix this.  
Just letting the file name automatically wrap doesn't look very good.

Maybe check the width of the terminal and check if the file name fits, if not, split the name manually and let it continue under where the name began

```
| Name: very-long-fi
|       le-name-that
|       wraps-severa
|       l-times
```

**Comment by asuender** [comment]
We can use the positions of any `_  -  .` or spaces to split the file names up, e.g.:
**Name:** ` my-long-filename.txt`

```
| Split it up in:
|                     my-
|                     long-
|                     filename.txt
```

**Comment by mfederczuk** [comment]
Could write a function to help with that:

```c++
std::vector<std::string> split_file_name(
    const std::string &file_name,
    unsigned int max_length,
    const std::string &wrap_chars = "_-.");
```

Of course we also gotta move the rest of the file infos down accordingly.
